src/controllers/createpdf.controller.ts

```diff
@@ -0,0 +1,98 @@
+import { Controller, Get, Res, Post, Body, Render, Param, Query } from '@nestjs/common';
+import { Response } from 'express';
+import { CreateUserData } from '../dtos/createpdf.userdata.dto';
+import { CreateLogInfo } from '../dtos/createpdf.loginfo.dto';
+import { CreateRunConf } from '../dtos/createpdf.runconf.dto';
+import { CreatePdfService } from '../services/createpdf.service';
+import { ApiUseTags } from '@nestjs/swagger';
+
+@ApiUseTags('createpdf')
+@Controller('createpdf')
+export class CreatepdfController {
+  constructor(public createPdfService: CreatePdfService) {}
+
+  /**
+   * Get Log information in pdf by id. /createpdf/getlog
+   * @param query there must be one id
+   * @param res response to client
+   */
+  @Get('getlog')
+  // query is needed, with body or param doesn't works
+  async getLogInfo(@Query() query: CreateLogInfo, @Res() res: Response): Promise<void> {
+    const epochTime = new Date();
+    let params;
+    // checking does id exists
+    try {
+      params = await this.createPdfService.getLogInfoParams(query.id);
+    } catch (err) {
+      return res.end('No such ID!');
+    }
+    return this.createPdfService
+    .createPdf(params, `Log_${query.id}_${epochTime.getTime()}.pdf`, '/templates/LogInfoTemplate.ejs', res);
+  }
+  /**
+   * Get differense between two runs in pdf
+   * by their id. /createpdf/getrunconf
+   * @param query there must be two id of runs
+   * @param res response to client
+   */
+  @Get('getrunconf')
+  // query is needed, with body or param doesn't works
+  async runconf(@Query() query: CreateRunConf, @Res() res: Response): Promise<void> {
+    const epochTime = new Date();
```

Comment:
The variable name is not correct since this object is a date object. Change it to
```typescript
const epochTime = (new Date).getTime();
```

Reply:
Yes, I know, but if I will write like that, lint will send an error

Reply:
I see, thanks for letting me know. I will take a look at it!

Reply:
It can be rewritten as
```typescript
const epochTime = new (Date)().getTime();
```